addons/onboarding/controllers/onboarding.py

```diff
@@ -12,11 +13,22 @@ def get_onboarding_data(self, route_name=None):
             return {}
 
         onboarding = request.env['onboarding.onboarding'].search([('route_name', '=', route_name)])
-        if onboarding and not onboarding._search_or_create_progress().is_onboarding_closed:
+        if onboarding and not self._safe_search_or_create_progress(onboarding).is_onboarding_closed:
             # JS implementation of the onboarding panel expects this data structure
             return {
                 'html': request.env['ir.qweb']._render(
                     'onboarding.onboarding_panel', onboarding._prepare_rendering_values())
             }
 
         return {}
+
+    @staticmethod
+    def _safe_search_or_create_progress(onboarding):
+        try:
+            current_progress = onboarding._search_or_create_progress()
+        except ValidationError as e:
+            # If initially not found but was created by another process before we tried to create
```

Comment:
Thank you for the review!
This commit is taken from another PR specifically on this (#109221), We should probably continue there. 

It  follows a temporary fix merged recently from #108183 because multiple records **were** created in a few databases. Although it's difficult to confirm, we've considered that these records were created if `_search_or_create` was called from multiple threads almost simultaneously => both could 

1. not find an existing record and 
2. try to create one (also considering that there is no table lock between the two?)

Then, only one of them will succeed and the other raise an error. 

If all this makes sense, we were hoping in that case to look again in the database, but what is it that we missing?
If one of our assumptions is broken, could you please tell us which one?

Reply:
The only exception that does trigger `retrying` are specifically LOCK_NOT_AVAILABLE, SERIALIZATION_FAILURE, DEADLOCK_DETECTED, all psycopg2 errors specific to serialization failure. I would recommend against trying some black rollbacking magic inside of the controller as anything done during the http routing (ir.http match/authenticate/pre_dispatch/frontend_pre_dispatch/dispatch) would be lost e.g. the `request.website` record would become unusable and could lead to other problems downstream.

Reply:
OK then, I'll try returning a specific flag and let the client decide whether to query again. 
Thank you both!

Reply:
The flag could be a [429 - Too many requests](https://httpwg.org/specs/rfc6585.html#status-429) http response. Not sure that the web client supports it though.

Reply:
I went with [503 - Service Unavailable](https://httpwg.org/specs/rfc9110.html#status.503) as the user shouldn't be aware that they are modifying anything and could/should just try again, there is nothing wrong with the request itself.

I would have liked to use something else than a dictionary for it, but as `Response`s end up as strings, they are less convenient. Maybe I missed something?
There is indeed a custom client handling as well, I didn't find anything else in owl.

(https://github.com/odoo/odoo/pull/109221/commits/7ec81f0a36b24d19a872a690ff97bbceb8c649fa)